pkg/splunk/handler_test.go

```diff
@@ -151,3 +158,39 @@ func TestSplunkHandlerBatching(t *testing.T) {
 		t.Fatalf("expected 1000 batches, got %d", stats.BatchCount)
 	}
 }
+
+func TestSlogtest(t *testing.T) {
+	messages := make(chan []byte)
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		buf, err := io.ReadAll(r.Body)
+		if err != nil {
+			t.Fatalf("read error: %v", err)
+		}
+		messages <- buf
+	}))
+	defer srv.Close()
+	defer close(messages)
+
+	c := SplunkConfig{
+		Level:              slog.LevelDebug,
+		URL:                srv.URL,
+		Source:             "s",
+		Hostname:           "h",
+		DefaultMaximumSize: 1, // force batching
+	}
+
+	slogtest.Run(t, func(t *testing.T) slog.Handler {
+		return NewSplunkHandler(context.Background(), c)
+	}, func(t *testing.T) map[string]any {
+		m := make(map[string]any)
+
+		buf := <-messages
+		err := json.Unmarshal(buf, &m)
+		if err != nil {
+			t.Fatalf("invalid json: %v\n%s", err, string(buf))
+		}
+
+		// unwrap the event
+		return m["event"].(map[string]any)["message"].(map[string]any)
+	})
+}
```

Comment:
This is a test from the standard library that performs about two dozens of various corner cases.